Clean up 'Contributing to Rust - Pull Requests' (#870)
rust-lang/rustc-dev-guide: src/contributing.md

to the end of the pull request description, and [@rust-highfive][rust-highfive] will assign
[@steveklabnik][steveklabnik] instead of a random person. This is entirely optional.

In addition to being reviewed by a human, pull requests are automatically tested
thanks to continuous integration (CI). Basically, every time you open and update

> a pull request, the CI builds the compiler and tests it against the

Reviewer: nit: CI is a process, not a noun
```suggestion
a pull request, CI builds the compiler and tests it against the
```
or
```suggestion
a pull request, Github Actions builds the compiler and tests it against the
```

Author: I think I've seen occurences of "the CI" in other parts of the guide... Not sure that's worth fixing, I'll just change that for my PR.

[compiler test suite][rctd], and also performs other tests such as checking that
your pull request is in compliance with Rust's style guidelines.

Running continuous integration tests allows PR authors to catch mistakes early
without going through a first review cycle, and also helps reviewers stay aware
of the status of a particular pull request.

Rust has plenty of CI capacity, and you should never have to worry about wasting
computational resources each time you push a change. It is also perfectly fine
(and even encouraged!) to use the CI to test your changes if it can help your

> productivity, e.g. if your machine is not very powerful.

Reviewer: This is true even for people with powerful machines ;) rustc is a bit of a hog.
```suggestion
productivity. In particular, we don't recommend running the full `x.py test` suite locally,
since it takes a very long time to execute.
```

After someone has reviewed your pull request, they will leave an annotation
on the pull request with an `r+`. It will look something like this:

    @bors r+

This tells [@bors], our lovable integration bot, that your pull request has
been approved. The PR then enters the [merge queue][merge-queue], where [@bors]
will run *all* the tests on *every* platform we support. If it all works out,
[@bors] will merge your code into `master` and close the pull request.

Depending on the scale of the change, you may see a slightly different form of `r+`:

    @bors r+ rollup

The additional `rollup` tells [@bors] that this change should always be "rolled up".
Changes that are rolled up are tested and merged alongside other PRs, to
speed the process up. Typically only small changes that are expected not to conflict
with one another are marked as "always roll up".

[rust-highfive]: https://github.com/rust-highfive
[steveklabnik]: https://github.com/steveklabnik
[@bors]: https://github.com/bors
[merge-queue]: https://buildbot2.rust-lang.org/homu/queue/rust

> ### Opening a PR

Reviewer: Hmm, I think I'd prefer for this to be a separate page to avoid making this one too long. But there's already a lot of sections, so maybe splitting them all up should be a separate PR?

Author: I think I could do that in a separate PR

You are now ready to file a pull request? Great! Here are a few points you
should be aware of.

All pull requests should be filed against the `master` branch, except in very
particular scenarios. Unless you know for sure that you should target another

> branch, `master` will be the right choice.

Reviewer: ```suggestion
branch, `master` will be the right choice (it's also the default).
```

Make sure your pull request is in compliance with Rust's style guidelines by running

>     $ ./x.py test tidy

Reviewer: No need to make them run it twice.
```suggestion
    $ ./x.py test tidy --bless
```

We recommand to make this check before every pull request (and every new commit
in a pull request); you can add [git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
before every push to make sure you never forget to make this check. The
CI will also run tidy and will fail if tidy fails.

Rust follows a _no merge-commit policy_, meaning, when you encounter merge
conflicts you are expected to always rebase instead of merging.  E.g. always use
rebase when bringing the latest changes from the master branch to your feature
branch.  Also, please make sure that fixup commits are squashed into other

> related commits with meaningful commit messages.

Reviewer: Once someone writes up #850, this should link there. But no need to solve two issues at once ;)

to the end of the pull request description, and [@rust-highfive][rust-highfive] will assign
[@steveklabnik][steveklabnik] instead of a random person. This is entirely optional.

In addition to being reviewed by a human, pull requests are automatically tested
thanks to continuous integration (CI). Basically, every time you open and update

> a pull request, CI builds the compiler and tests it against the

Reviewer 2: ```suggestion
a PR, CI builds the compiler and tests it against the
```

Reviewer 2: This avoids having to decide between *pull request* and *Pull Request*

Author: I don't think Pull Request vs pull request is a problem however

Reviewer 2: The inconsistency bothered me

[@bors] will merge your code into `master` and close the pull request.

Depending on the scale of the change, you may see a slightly different form of `r+`:

    @bors r+ rollup

The additional `rollup` tells [@bors] that this change should always be "rolled up".
Changes that are rolled up are tested and merged alongside other PRs, to
speed the process up. Typically only small changes that are expected not to conflict
with one another are marked as "always roll up".

[rust-highfive]: https://github.com/rust-highfive
[steveklabnik]: https://github.com/steveklabnik
[@bors]: https://github.com/bors
[merge-queue]: https://buildbot2.rust-lang.org/homu/queue/rust

### Opening a PR

You are now ready to file a pull request? Great! Here are a few points you
should be aware of.

All pull requests should be filed against the `master` branch, except in very
particular scenarios. Unless you know for sure that you should target another
branch, `master` will be the right choice (it's also the default).

Make sure your pull request is in compliance with Rust's style guidelines by running

    $ ./x.py test tidy --bless

> We recommand to make this check before every pull request (and every new commit

Reviewer 2: ```suggestion
We recommend to make this check before every pull request (and every new commit
```
